src/data-model-providers/codegen/CodegenDataModelProvider_Write.cpp

```diff
@@ -94,6 +94,16 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::WriteAttribute(const Dat
 {
     if (auto * cluster = mRegistry.Get(request.path); cluster != nullptr)
     {
+        if (request.path.mDataVersion.HasValue())
```

Comment:
API docs for Provider.h say:

```
    /// Requests a write of an attribute.
    ///
    /// When this is invoked, caller is expected to have already done some validations:
    ///    - cluster `data version` has been checked for the incoming request if applicable
```

I don't see that happening right now, but it's the right behavior and we should fix API consumers to do the right thing, no?